portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -36,6 +42,15 @@ public void init() {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    
+    Query query = new Query("Comments").addSort("timestamp", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    for (Entity entity : results.asIterable()) {
+        String comment = (String) entity.getProperty("text");
+        comments.add(comment);
```

Comment:
Do you have a lifetime issue here? What happens if you rapidly refresh the page? (I'd guess: duplicated comments)

Now that you have these in a datastore, do you still need the "comments" var at class scope?

Reply:
Comments being at the class scope definitely has that effect. Fixed it.